benches/msm_throughput.rs

```diff
@@ -0,0 +1,61 @@
+//! Benchmarks multiscalar multiplication throughput as executed via the `Executor`
+
+use std::time::{Duration, Instant};
+
+use criterion::{black_box, criterion_group, criterion_main, BenchmarkId, Criterion, Throughput};
+use itertools::Itertools;
+use mpc_stark::{
+    algebra::authenticated_stark_point::AuthenticatedStarkPointResult,
+    test_helpers::execute_mock_mpc,
+};
+use tokio::runtime::Builder as RuntimeBuilder;
+
+/// Measure the throughput and latency of a variable-sized MSM
+pub fn bench_msm_throughput(c: &mut Criterion) {
+    let runtime = RuntimeBuilder::new_multi_thread()
+        .worker_threads(3)
+        .enable_all()
+        .build()
+        .unwrap();
+
+    let mut group = c.benchmark_group("msm-throughput");
+    for circuit_size in [100, 1000, 10000].into_iter() {
+        group.throughput(Throughput::Elements(circuit_size as u64));
+        group.bench_function(BenchmarkId::from_parameter(circuit_size), |b| {
+            let mut b = b.to_async(&runtime);
+            b.iter_custom(|n_iters| async move {
+                let mut total_time = Duration::from_millis(0);
+                for _ in 0..n_iters {
+                    let (elapsed1, elapsed2) = execute_mock_mpc(|fabric| async move {
+                        let scalars = (0..circuit_size)
+                            .map(|_| fabric.one_authenticated())
+                            .collect_vec();
+                        let points = (0..circuit_size)
+                            .map(|_| fabric.curve_identity_authenticated())
+                            .collect_vec();
```

Comment:
I don't have a great intuition here, but I think it would be a more accurate benchmark if we used random scalars & the curve generator. using `1` and the identity respectively doesn't result in accurately complex ec operations, does it? I'm sure a scalar mul is, for our intents & purposes, a ~constant time operation, but this would let us get a realistic constant